src/index.ts

```diff
@@ -1,7 +1,8 @@
 import { TreasureMapBot } from "./bot";
 import { askAndParseEnv, identity, requireEnv } from "./lib";
 
-async function main() {
+// call an async function immediately
```

Comment:
I think this is unnecessary, this makes the code less readable and does not change at the behavior of the async function. There is no performance difference between this and calling `main` function on the next line. If you truly which to call the function immediately, something like `setImmediate` or `process.nextTick`  would be useful, but it is not useful in this case.

If you want some information, I suggest this article https://nodejs.dev/learn/understanding-setimmediate

Reply:
I know it is a work in progress, so, take my words **lightly**. Thanks for the intent to contribute.